sycl/include/sycl/queue.hpp

```diff
@@ -480,11 +480,14 @@ class __SYCL_EXPORT queue : public detail::OwnerLessBase<queue> {
   /// trivially copyable.
   /// \param Count is the number of times to fill Pattern into Ptr.
   /// \return an event representing fill operation.
-  template <typename T> event fill(void *Ptr, const T &Pattern, size_t Count) {
-    // TODO: to add code location as parameter when ABI break is permitted
-    const detail::code_location CodeLoc("sycl/queue.hpp", "fill", 0, 0);
-    return submit([&](handler &CGH) { CGH.fill<T>(Ptr, Pattern, Count); },
-                  CodeLoc);
+  template <typename T>
+  event fill(void *Ptr, const T &Pattern,
+             size_t Count _CODELOCPARAM(&CodeLoc)) {
+    detail::tls_code_loc_t TlsCodeLocCapture(CodeLoc);
```

Comment:
_CODELOCPARAM(&CodeLoc)  must be paired with:

_CODELOCARG(&CodeLoc);  between line 485 & 486:

```suggestion
    template <typename T>
    event fill(void *Ptr, const T &Pattern,
             size_t Count _CODELOCPARAM(&CodeLoc)) {
    _CODELOCARG(&CodeLoc); 
    detail::tls_code_loc_t TlsCodeLocCapture(CodeLoc);
```

If DISABLE_SYCL_INSTRUMENTATION_METADATA is enabled, then the code will not compile due to undefined variable. See proper use in other constructs such as [submit](https://github.com/intel/llvm/blob/0b46dd62da31aed14679bf3c9bc532d693413c66/sycl/include/sycl/queue.hpp#L308)